dlc-manager/src/chain_monitor.rs

```diff
@@ -108,41 +107,91 @@ impl ChainMonitor {
         }
     }
 
-    pub(crate) fn process_block(
-        &self,
-        block: &Block,
-        height: u64,
-    ) -> Vec<(Transaction, ChannelInfo)> {
-        let mut res = Vec::new();
-
+    /// Check if any watched transactions are part of the block, confirming them if so.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the new block's height is not exactly one more than the last processed height.
+    pub(crate) fn process_block(&mut self, block: &Block, height: u64) {
         assert_eq!(self.last_height + 1, height);
 
-        for tx in &block.txdata {
-            let channel_info = self.watched_tx.get(&tx.txid()).or_else(|| {
-                for txid in tx.input.iter().map(|x| &x.previous_output.txid) {
-                    let info = self.watched_tx.get(txid);
-                    if info.is_some() {
-                        return info;
-                    }
-                }
-                None
-            });
-            if let Some(channel_info) = channel_info {
-                res.push((tx.clone(), channel_info.clone()));
+        for (txid, state) in self.watched_tx.iter_mut() {
```

Comment:
Since `watched_tx` is a hash map, it feels like it would be more efficient to iterate the tx in the block and check if they are watched ones. Now it will most of the time iterate all block transaction many times.

Reply:
I think you are totally right that it makes way more sense to iterate through the block to take advantage of the constant-time lookup in the hashmap.

I guess it does depend on the size of the hashmap compared to the size of the block, but I don't think we need to worry about that now. I'll go back to the original!